### PR TITLE
[flink] support order sort in flink compact action

### PR DESCRIPTION
**Pull request by leaves12138**
Support order sort in flink action

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/InternalRowType.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.apache.paimon.shade.guava30.com.google.common.base.Objects;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+
+import java.util.Arrays;
+
+/** A {@link TypeInformation} for internal row serializer. */
+public class InternalRowType extends TypeInformation<InternalRow> {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/InternalRowType.java:33]: Can you just use `InternalTypeInfo`?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java [hunk 2]
```diff
@@ -66,6 +66,13 @@ public SortCompactAction(
     public void run() throws Exception {
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setRuntimeMode(RuntimeExecutionMode.BATCH);
+
+        // we enable object reuse, we copy the un-reusable object ourselves.
+        env.getConfig().enableObjectReuse();
+
+        // kryo serializer is not good
+        env.getConfig().disableGenericTypes();
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java:74]: Why to disable generic type?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java [hunk 3]
```diff
@@ -66,6 +66,13 @@ public SortCompactAction(
     public void run() throws Exception {
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setRuntimeMode(RuntimeExecutionMode.BATCH);
+
+        // we enable object reuse, we copy the un-reusable object ourselves.
+        env.getConfig().enableObjectReuse();
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java:71]: Move this into `ActionBase`.

And use env in `ActionBase`.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java [hunk 4]
```diff
@@ -66,6 +66,13 @@ public SortCompactAction(
     public void run() throws Exception {
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setRuntimeMode(RuntimeExecutionMode.BATCH);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java:68]: remove this, batch or stream can be set by users using Flink CLI.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java (hunk 3 above)
Comment (leaves12138) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java]: done

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java (hunk 4 above)
Comment (leaves12138) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/SortCompactAction.java]: batch mode may need to forcely specified

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/InternalRowType.java (hunk 1 above)
Comment (leaves12138) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/utils/InternalRowType.java]: done

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**